### DISCUSSION TITLE
Flags `toString` is a little dangerous

### DISCUSSION THREAD

**Discussion by opotowsky** [terrapower/armi#1911] [General]
Converting unordered flags to a string may or may not be dangerous. A downstream repo certainly is using it in a dangerous way. Here's the code:

https://github.com/terrapower/armi/blob/96a5c88d6167cee989378599b6d341c351c291bd/armi/reactor/flags.py#L180-L188

Maybe this is a #wontfix thing and people just need to not do dangerous things. But I wanted to demonstrate how this could be used in a dangerous way. 

Say you want to get a parameter, `rxFuelAxialExpansionCoeffPerTemp`. You have the following info: `type = "Fuel"` and `flags = AXIAL | EXPANSION`. 

In python 3.9 when you call `flags.toString()`, you get `"Axial Expansion"`. You can remove the space and use that to construct your parameter name. In python 3.11, you get `"Expansion Axial"`, in which case you need to know ahead of time to flip the flags before moving forward.

**Comment by john-science** [comment]
Your "expansion axial" example there feels like it should be caught by our unit tests. We run the same tests in Python 3.9 - 3.12, you'd think we would have noticed this.

Perhaps we need to add a test?

**Comment by john-science** [comment]
Wait, is someone calling the private `_to String()` outside ARMI?

**Reply by jakehader** [reply]
No, https://github.com/terrapower/armi/blob/96a5c88d6167cee989378599b6d341c351c291bd/armi/reactor/flags.py#L304 is being called.

**Comment by opotowsky** [comment]
@jakehader shared this, which might explain it
![image](https://github.com/user-attachments/assets/28d57a33-59d5-4190-a476-2638113eeb02)

But @john-science was writing a unit test and found the ordering is not guaranteed in py39. So the `toString` method will likely return an alphabetized version. 

I would argue the `toString` method should only be used on individual flags, personally. So it should only accept a single flag. Then we don't need to worry about ordering at all.

**Comment by john-science** [comment]
We just found out that `Flags.toString()` in Python 3.9, for multiple flags, returns them in random order every time. There is currently NO guarantee about flag ordering from this method.

I don't like that, so I'll just fix it to be alphabetical. Consistency is better than randomness.

**Reply by keckler** [reply]
Did this get turned into an issue? I do not see one, but am just wondering if this is actually something that will be done or not?

**Reply by john-science** [reply]
The PR is up: https://github.com/terrapower/armi/pull/1912
